File: test_conversion.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import requests
import time
import os
import json

def test_conversion():
    # Путь к файлу для конвертации
    input_file = r"C:\Users\Denis\Desktop\Convertio\video for test\Screenshot (19).png"
    output_dir = r"C:\Users\Denis\Desktop\Convertio\video for test"
    
    print(f"Тестирование конвертации: {input_file}")
    print(f"Результат будет сохранен в: {output_dir}")
    
    # Проверяем существование файла
    if not os.path.exists(input_file):
        print(f"Файл не найден: {input_file}")
        return
    
    # Получаем размер файла
    file_size = os.path.getsize(input_file)
    print(f"Размер файла: {file_size / 1024:.2f} KB")
    
    try:
        # Шаг 1: Запускаем конвертацию
        print("\n🚀 Шаг 1: Запуск конвертации...")
        
        with open(input_file, 'rb') as f:
            files = {'file': f}
            data = {'outputformat': 'jpg'}
            
            response = requests.post(
                'http://localhost:3002/api/start-conversion',
                files=files,
                data=data,
                timeout=30
            )
        
        if response.status_code != 200:
            print(f"❌ Ошибка запуска конвертации: {response.status_code}")
            print(f"Response: {response.text}")
            return
        
        result = response.json()
        job_id = result['id']
        print(f"✅ Конвертация запущена! Job ID: {job_id}")
        
        # Шаг 2: Отслеживаем статус
        print("\n⏳ Шаг 2: Отслеживание статуса...")
        
        max_attempts = 30
        for attempt in range(max_attempts):
            print(f"📊 Попытка {attempt + 1}/{max_attempts}...")
            
            status_response = requests.get(
                f'http://localhost:3002/api/conversion-status/{job_id}',
                timeout=10
            )
            
            if status_response.status_code != 200:
                print(f"⚠️ Ошибка получения статуса: {status_response.status_code}")
                time.sleep(2)
                continue
            
            status_data = status_response.json()
            print(f"Status: {status_data.get('status')}, Step: {status_data.get('step')}")
            
            if status_data.get('status') == 'finished' and status_data.get('output'):
                # Шаг 3: Скачиваем результат
                download_url = status_data['output']['url']
                print(f"\n✅ Конвертация завершена! Скачиваем файл...")
                print(f"🔗 URL: {download_url}")
                
                # Скачиваем конвертированный файл
                download_response = requests.get(download_url, timeout=30)
                if download_response.status_code == 200:
                    output_file = os.path.join(output_dir, "Screenshot_19_converted.jpg")
                    
                    with open(output_file, 'wb') as f:
                        f.write(download_response.content)
                    
                    print(f"💾 Файл сохранен: {output_file}")
                    print(f"📊 Размер результата: {len(download_response.content) / 1024:.2f} KB")
                    print("🎉 Конвертация успешно завершена!")
                    return
                else:
                    print(f"❌ Ошибка скачивания: {download_response.status_code}")
                    return
            
            elif status_data.get('status') == 'error':
                print(f"❌ Ошибка конвертации: {status_data}")
                return
            
            # Ждем перед следующей проверкой
            time.sleep(3)
        
        print("⏰ Превышен лимит времени ожидания")
        
    except requests.exceptions.RequestException as e:
        print(f"🌐 Ошибка сети: {e}")
    except Exception as e:
        print(f"💥 Неожиданная ошибка: {e}")

if __name__ == "__main__":
    test_conversion()

File: test_conversion.js
const axios = require('axios');
const fs = require('fs');
require('dotenv').config();

async function testConversion() {
    const cloudConvertKey = process.env.CLOUDCONVERT_KEY;
    
    if (!cloudConvertKey) {
        console.error('CLOUDCONVERT_KEY not set');
        return;
    }

    try {
        console.log('🧪 Testing CloudConvert API directly...');
        
        // Читаем тестовый файл
        const fileBuffer = fs.readFileSync('test.png');
        const base64File = fileBuffer.toString('base64');
        
        console.log(`📄 File size: ${fileBuffer.length} bytes`);
        console.log('🔄 Creating CloudConvert job...');
        
        // Создаем job в CloudConvert
        const response = await axios.post('https://api.cloudconvert.com/v2/jobs', {
            tasks: {
                'import-file': {
                    operation: 'import/base64',
                    file: base64File,
                    filename: 'test.png'
                },
                'convert-file': {
                    operation: 'convert',
                    input: 'import-file',
                    output_format: 'jpg'
                },
                'export-file': {
                    operation: 'export/url',
                    input: 'convert-file'
                }
            }
        }, {
            headers: {
                'Authorization': `Bearer ${cloudConvertKey}`,
                'Content-Type': 'application/json'
            }
        });

        const jobId = response.data.data.id;
        console.log(`✅ Job created successfully: ${jobId}`);
        
        // Ждем завершения конвертации
        let jobStatus = 'waiting';
        let attempts = 0;
        const maxAttempts = 30;
        
        while (jobStatus !== 'finished' && jobStatus !== 'error' && attempts < maxAttempts) {
            await new Promise(resolve => setTimeout(resolve, 3000)); // Ждем 3 секунды
            
            const statusResponse = await axios.get(`https://api.cloudconvert.com/v2/jobs/${jobId}`, {
                headers: {
                    'Authorization': `Bearer ${cloudConvertKey}`
                }
            });
            
            jobStatus = statusResponse.data.data.status;
            attempts++;
            
            console.log(`📊 Status check ${attempts}: ${jobStatus}`);
        }
        
        if (jobStatus === 'finished') {
            const finalJob = await axios.get(`https://api.cloudconvert.com/v2/jobs/${jobId}`, {
                headers: {
                    'Authorization': `Bearer ${cloudConvertKey}`
                }
            });
            
            const exportTask = finalJob.data.data.tasks.find(task => task.operation === 'export/url');
            if (exportTask && exportTask.result && exportTask.result.files) {
                const downloadUrl = exportTask.result.files[0].url;
                const fileSize = exportTask.result.files[0].size;
                
                console.log(`🎉 Conversion completed!`);
                console.log(`📥 Download URL: ${downloadUrl}`);
                console.log(`📄 File size: ${fileSize} bytes`);
                
                // Скачиваем сконвертированный файл
                console.log('💾 Downloading converted file...');
                const fileResponse = await axios.get(downloadUrl, { 
                    responseType: 'arraybuffer' 
                });
                
                fs.writeFileSync('test_converted.jpg', fileResponse.data);
                console.log('✅ File saved as test_converted.jpg');
                
                return {
                    success: true,
                    jobId: jobId,
                    downloadUrl: downloadUrl,
                    fileSize: fileSize,
                    localFile: 'test_converted.jpg'
                };
            }
        } else if (jobStatus === 'error') {
            console.error('❌ Job failed');
            return { success: false, error: 'Job failed' };
        } else {
            console.error('⏱️ Job timed out');
            return { success: false, error: 'Timeout' };
        }
        
    } catch (error) {
        console.error('🚨 Error:', error.response ? error.response.data : error.message);
        return { success: false, error: error.message };
    }
}

// Запуск теста
testConversion().then(result => {
    console.log('\n📋 Final result:', result);
    process.exit(result.success ? 0 : 1);
});